presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -945,6 +953,48 @@ void PrestoServer::addServerPeriodicTasks() {
         ttlCheckInterval,
         "cache_ttl");
   }
+
+  if (isCachePeriodicFullPersistenceEnabled()) {
+    PRESTO_STARTUP_LOG(INFO)
+        << "Initializing cache periodic full persistence task...";
+    auto* cache = velox::cache::AsyncDataCache::getInstance();
+    VELOX_CHECK_NOT_NULL(cache);
+    auto* ssdCache = cache->ssdCache();
+    VELOX_CHECK_NOT_NULL(ssdCache);
+    const auto* systemConfig = SystemConfig::instance();
+    const int64_t cacheFullPersistenceIntervalUs =
+        std::chrono::duration_cast<std::chrono::microseconds>(
+            systemConfig->asyncCacheFullPersistenceInterval())
+            .count();
+    const auto asyncCacheSsdCheckpointGb =
+        systemConfig->asyncCacheSsdCheckpointGb();
+    periodicTaskManager_->addTask(
+        [asyncCacheSsdCheckpointGb, cache, ssdCache]() {
+          try {
+            if (!ssdCache->startWrite()) {
+              return;
+            }
+            LOG(INFO) << "Persisting full cache to SSD...";
+            cache->saveToSsd(true);
+            ssdCache->waitForWriteToFinish();
+
+            if (asyncCacheSsdCheckpointGb == 0) {
+              return;
+            }
+
+            if (!ssdCache->startWrite()) {
+              return;
+            }
+            ssdCache->checkpoint();
+            ssdCache->waitForWriteToFinish();
+            LOG(INFO) << "Cache full persistence completed.";
+          } catch (const VeloxRuntimeError& e) {
+            LOG(ERROR) << "Failed to persistent cache to SSD: " << e.what();
+          }
```

Comment:
Do we expect other exception types? E.g. exceptions thrown by os when writing and etc. Do we want to have a wild catch at the end? Thanks